rootio/tchain_test.go

```diff
@@ -246,3 +246,55 @@ func BenchmarkChainTreeScannerStruct(b *testing.B) {
 		b.Fatal(err)
 	}
 }
+
+func TestSeekEntry(t *testing.T) {
+	files := []string{
+		"testdata/chain.1.root",
+		"testdata/chain.2.root",
+	}
+
+	trees := make([]rootio.Tree, len(files))
+	for i, fname := range files {
+		f, err := rootio.Open(fname)
+		if err != nil {
+			t.Fatalf("could not open ROOT file %q: %v", fname, err)
+		}
+		defer f.Close()
+
+		obj, err := f.Get("tree")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		trees[i] = obj.(rootio.Tree)
+	}
+	chain := rootio.Chain(trees...)
+
+	type Data struct {
+		Event struct {
+			Beg       string      `rootio:"Beg"`
+			F64       float64     `rootio:"F64"`
+			ArrF64    [10]float64 `rootio:"ArrayF64"`
+			N         int32       `rootio:"N"`
+			SliF64    []float64   `rootio:"SliceF64"`
+			StdStr    string      `rootio:"StdStr"`
+			StlVecF64 []float64   `rootio:"StlVecF64"`
+			StlVecStr []string    `rootio:"StlVecStr"`
+			End       string      `rootio:"End"`
+		} `rootio:"evt"`
+	}
+
+	sc, err := rootio.NewTreeScanner(chain, &Data{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer sc.Close()
+
+	var entries = []int64{1, 0, 2, 9, 10, 11, 0, 19}
+	for _, entry := range entries {
+		err := sc.SeekEntry(entry)
```

Comment:
ok, but we should also test that we landed on the correct entry.
_e.g._: we know that the field `F64` should be equal to `float64(entry)`.